python/lsst/pipe/tasks/loadReferenceCatalog.py

```diff
@@ -0,0 +1,380 @@
+            self._determineFluxFields(center, filterList)
+
+        skyCircle = self.refObjLoader.loadSkyCircle(center, radius,
+                                                    self._referenceFilter,
+                                                    epoch=epoch)
+
+        if not skyCircle.refCat.isContiguous():
+            refCat = skyCircle.refCat.copy(deep=True)
+        else:
+            refCat = skyCircle.refCat
+
+        return self._formatCatalog(refCat, filterList)
+
+    def _formatCatalog(self, refCat, filterList):
+        """Format a reference afw table into the final format.
+
+        This method applies reference selections and color terms as specified
+        by the config.
+
+        Parameters
+        ----------
+        refCat : `lsst.afw.table.SourceCatalog`
+            Reference catalog in afw format.
+        filterList : `list` [`str`]
+            List of camera physicalFilter names to apply color terms.
+
+        Returns
+        -------
+        refCat : `numpy.ndarray`
+            Reference catalog.
+        """
+        if self.config.doReferenceSelection:
+            goodSources = self.referenceSelector.selectSources(refCat)
+            selected = goodSources.selected
+        else:
+            selected = np.ones(len(refCat), dtype=bool)
+
+        npRefCat = np.zeros(np.sum(selected), dtype=[('ra', 'f8'),
+                                                     ('dec', 'f8'),
+                                                     ('refMag', 'f4', (len(filterList), )),
+                                                     ('refMagErr', 'f4', (len(filterList), ))])
+
+        if npRefCat.size == 0:
+            # Return an empty catalog if we don't have any selected sources.
+            return npRefCat
+
+        # Natively "coord_ra" and "coord_dec" are stored in radians.
+        # Doing this as an array rather than by row with the coord access is
+        # approximately 600x faster.
+        npRefCat['ra'] = np.rad2deg(refCat['coord_ra'][selected])
+        npRefCat['dec'] = np.rad2deg(refCat['coord_dec'][selected])
+
+        # Default (unset) values are 99.0
+        npRefCat['refMag'][:, :] = 99.0
+        npRefCat['refMagErr'][:, :] = 99.0
+
+        if self.config.doApplyColorTerms:
+            if isinstance(self.refObjLoader, ReferenceObjectLoader):
+                # Gen3
+                refCatName = self.refObjLoader.config.value.ref_dataset_name
+            else:
+                # Gen2
+                refCatName = self.refObjLoader.ref_dataset_name
+
+            for i, (filterName, fluxField) in enumerate(zip(self._fluxFilters, self._fluxFields)):
+                if fluxField is None:
+                    # There is no matching reference band.
+                    # This will leave the column filled with 99s
+                    continue
+                self.log.debug(f"Applying color terms for filtername={filterName}")
```

Comment:
This should probably be camelCase `filterName`, to match the rest of the script. Also, a nice shortcut here to avoid typing out the object name twice would be to use `self.log.debug(f"Applying color terms for {filterName=}")` (which also adds single-quotes around the string output too).

Reply:
That's handy!  But on another ticket I was admonished (again) that for logging (especially debug) I should be using the `%s` formatting for lazy-evaluation.  (I doubt it matters much for performance, but it's the "right" way to log).

Reply:
Ah, I wasn't aware that there was a 'correct' way to log debug data, thank you for letting me know! In which case, the proposal you suggest sounds good.